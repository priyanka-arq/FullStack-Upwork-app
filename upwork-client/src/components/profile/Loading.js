import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";

const useStyles = makeStyles((theme) => ({
  root: {
    margin: "auto",
    "& > * + *": {
      marginLeft: theme.spacing(1),
    },
  },
  hide: {
    display: "none",
  },
  green: {
    color: "green",
  },
}));

export default function Loading(props) {
  const classes = useStyles();
  return (
    <div className={props.show ? classes.root : classes.hide}>
      <CircularProgress color="secondary" />
      <CircularProgress className={classes.green} />
      <CircularProgress />
    </div>
  );
}
