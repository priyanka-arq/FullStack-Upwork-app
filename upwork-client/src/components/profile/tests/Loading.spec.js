import React from "react";
import { shallow } from "enzyme";
import Loading from "../Loading";
import toJSON from "enzyme-to-json";
import CircularProgress from "@material-ui/core/CircularProgress";
let wrapped;

beforeEach(() => {
  wrapped = shallow(<Loading />);
});

afterEach(() => {
  //unmount component from dom structure, so that it won't interfer with other component
  wrapped.unmount();
});

test("Should CircularProgress correctly ", () => {
  expect(toJSON(wrapped)).toMatchSnapshot();
});

it("show profile card", () => {
  const wrapped = shallow(<Loading />);

  //find will return an array
  //will test how many time <CircularProgress /> component is render inside a Loading component
  expect(wrapped.find(CircularProgress).length).toEqual(3);
});
