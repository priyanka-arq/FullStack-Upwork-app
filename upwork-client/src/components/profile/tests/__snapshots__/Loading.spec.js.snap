// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CircularProgress correctly  1`] = `
<div
  className="makeStyles-hide-2"
>
  <WithStyles(ForwardRef(CircularProgress))
    color="secondary"
  />
  <WithStyles(ForwardRef(CircularProgress))
    className="makeStyles-green-3"
  />
  <WithStyles(ForwardRef(CircularProgress)) />
</div>
`;
